src/github.com/matrix-org/dendrite/roomserver/api/query.go

```diff
@@ -0,0 +1,96 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/matrix-org/gomatrixserverlib"
+	"net/http"
+)
+
+// StateKeyTuple is a pair of an event type and state_key.
+type StateKeyTuple struct {
+	EventType     string
+	EventStateKey string
```

Comment:
Given the significance of `""` I don't like the type of this being `string`, because if people forget to specify the `EventStateKey` it will default to `""` rather than doing what I think it should be doing, which is failing.